Exiled.Events/EventArgs/Scp3114/DancingEventArgs.cs

```diff
@@ -0,0 +1,59 @@
+// -----------------------------------------------------------------------
+// <copyright file="DancingEventArgs.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.EventArgs.Scp3114
+{
+    using Exiled.API.Enums;
+    using Exiled.API.Features;
+    using Exiled.API.Features.Roles;
+    using Exiled.Events.EventArgs.Interfaces;
+    using Exiled.Events.Patches.Events.Scp3114;
+
+    /// <summary>
+    /// Contains all information before SCP-3114 starts or stops dancing.
+    /// </summary>
+    public class DancingEventArgs : IDeniableEvent, IScp3114Event
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DancingEventArgs"/> class.
+        /// </summary>
+        /// <param name="newState"><inheritdoc cref="DanceType"/></param>
+        /// <param name="player"><inheritdoc cref="Player"/></param>
+        /// <param name="isAllowed"><inheritdoc cref="IsAllowed"/></param>
+        public DancingEventArgs(bool newState, Player player, bool isAllowed = true)
+        {
+            Player = player;
+            Scp3114 = player.Role.As<Scp3114Role>();
+            IsDancing = newState;
+            DanceType = (DanceType)(newState ? UnityEngine.Random.Range(0, Scp3114.Dance._danceVariants) : 0);
```

Comment:
Maybe add None in enum for this case? I think there can be an error cuz enum doesn't have such value (or it has, but its Dance1)

Reply:
I would like to give name to the different dance but yeah 0 will be None

Reply:
I also don't know to name dances. I don't know them. maybe we should leave a small description about dance and name it based on it's style?